plugins/org.eclipse.n4js.utils.ui/src/org/eclipse/n4js/utils/ui/editor/AvoidRefreshDocumentProvider.java

```diff
@@ -0,0 +1,30 @@
+/**
+ * Copyright (c) 2017 NumberFour AG.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   NumberFour AG - Initial API and implementation
+ */
+package org.eclipse.n4js.utils.ui.editor;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IProgressMonitor;
+import org.eclipse.xtext.ui.editor.model.XtextDocumentProvider;
+
+/**
+ * This class temporarily solves the problem that a file is refreshed when opened.
```

Comment:
It's not temporarily. The class will - if its written like it is - always do exactly that :)

Reply:
By "temporarily", I actually mean that ideally this problem would be fixed in the Xtext framework. But I will remove "temporarily" :)